src/controllers/user.controller.ts

```diff
@@ -389,17 +389,31 @@ export const acceptGroupInvitation = async (req: Request, res: Response) => {
   }
 
   try {
-    // Remove groupId from invited groups and add to currentGroup
-    const targetUser = await User.findByIdAndUpdate(userId, {
-      $pull: { invitedGroups: groupId },
-      currentGroup: groupId,
-    });
+    const targetUser = await User.findById(userId);
 
     // Check if the user exists
     if (targetUser === null) {
       return res.status(400).send({ message: 'User does not exist!' });
     }
 
+    // Check if the user is already in a group
+    if (targetUser?.currentGroup) {
+      return res
+        .status(400)
+        .send({ message: 'User already has a current group!' });
+    }
+
+    // Remove groupId from invited groups and add to currentGroup
+    targetUser.invitedGroups = targetUser.invitedGroups.filter(
+      group => group._id.toString() !== groupId
+    );
+
+    // Add groupId to currentGroup
+    targetUser.currentGroup = new mongoose.Types.ObjectId(groupId);
+
+    // Remove groupId from invited groups and add to currentGroup
+    await targetUser.save();
+
```

Comment:
This should only be done after we have verified that `groupId` exists. So, in case `targetGroup` doesn't actually exist, the target user's fields aren't updated.

Also, if the `groupId` does exist, we should still remove it from the target user's invited groups.